packages/@react-types/shared/src/collections.d.ts

```diff
@@ -49,6 +50,8 @@ export interface SectionProps<T> {
   items?: Iterable<T>
 }
 
+export interface SelectionGroupProps<T> extends Omit<CollectionBase<T>, 'disabledKeys'>, Omit<MultipleSelection, "disabledKeys"> {}
```

Comment:
We can allow passing disabledKeys to SelectionGroup but then we'll have to add a way to set disabled keys state in parent's selection manager as it controls the keyboard navigation. Let me know If there's another way!